src/3_PB/SpecificaTecnica/content.typ

```diff
@@ -0,0 +1,653 @@
+
+== Configurazione visualizzazione e sistema di allerta
+=== Dashboard
+La visualizzazione dei dati attraverso #glossary[pannelli] si compone delle seguenti #glossary[dashboard] di #glossary[Grafana]. Le configurazioni delle #glossary[dashboard] vengono salvate in formato #glossary[JSON], il che permette un facile sviluppo e manutenzione. I dati per la visualizzazione sono prelevati dalle tabelle di #glossary[ClickHouse] tramite query alle tabelle #glossary[time series] e di dati agggregati. Di seguito vengono descritte le #glossary[dashboard] per il visualization layer:
+- *Ambientale*: visualizzazione dei dati relativi a temperatura, umidità, inquinamento dell'aria, vento e precipitazioni atmosferiche, tramite grafici a linee, mappe e indicatori; include inoltre una mappa che illustra la posizione dei sensori e il loro tipo, rappresentato con un colore diverso;
+- *Urbanistica*: visualizzazione dei dati relativi a parcheggi, colonne di ricarica, zone ecologiche, livello di congestione stradale e batteria delle biciclette elettriche, tramite mappe ed indicatori;
+- *Dati grezzi*: visualizzazione dei dati grezzi provenienti dai sensori, tramite una tabella che mostra i dati in tempo reale, con la possibilità di filtrare per specifico #glossary[sensore] e per tipo;
+- *Superamento soglie*: visualizzazione dei dati relativi al superamento delle soglie di allerta per i diversi tipi di sensori; le allerte vengono visualizzate tramite una tabella che mostra il tipo di #glossary[sensore], il valore rilevato e il timestamp.
+
+=== Sistema di notifica
+#glossary[Grafana] offre un robusto sistema di allerta che consente di rilevare e reagire prontamente a condizioni anomale o critiche nei dati monitorati. Questo sistema non solo permette di definire regole di allerta basate su query ai data source, ma anche di personalizzare le notifiche inviate attraverso vari canali tra i quali Discord.
+Le regole di allerta sono il cuore di questo sistema, consentendo di definire le condizioni che devono attivare un'allerta. Queste regole possono essere configurate per una vasta gamma di scenari; nel nostro contesto di applicazione vengono implementate per rilevare il superamento di determinate soglie, come ad esempio per rilevare temperature eccessive o livelli di inquinantamento atmosferico pericolosi.
+Una volta che un'allerta è attivata, passa attraverso tre stati distinti:
+
+- in attesa (pending): indicando che l'allerta è stata innescata ma la sua conferma è ancora in corso. Questo stato è tipicamente utilizzato per regolare l'invio delle notifiche, assicurandosi che l'allerta sia stabile prima di comunicare il problema;
+
+- attiva (firing): l'allerta è stata confermata e la condizione critica è stata verificata, di conseguenza vengono inviate le notifiche ai canali configurati;
+
+- ok: l'allerta è stata risolta e la situazione è tornata alla normalità.
+
```

Comment:
```suggestion 
Nel realizzare il sistema di notifica il team ha optato per mantenerlo il più semplice possibile andando di fatto a rimuovere lo stato di pending in modo tale che ogni allerta venisse notificata appena rilevata. 
Inoltre si è valutato con preferibile rimuovere le notifiche relative al condizione di ok. 
Ogni allerta viene definita all'interno di un "allert group" a se stante, in modo tale che le regole siano eseguite in contemporanea. Ogni regola verifica, a intervalli regolati di 5 minuti, se la condizione di superamento della soglia impostata si è verificata nei 5 minuti precedenti, e relativamente ad ogni sensore trattiene il valore massimo che abbia superato tale soglia.

Reply:
Una nota riguardo a questa aggiunta, non saprei se inserirlo a questo punto del testo, vedi un pò te se vuoi modificare il flow del testo mattendolo o prima o dopo, inoltre non so se avrebbe senso andare a mettere anche la query sql  e le trasformazioni (? forse goofy)

Reply:
oppure anche iserire sta roba in coda a quello che avevi già fatto tu, in modo tala che il tuo fosse una presentazione degli strumenti e la parte nuova più come l'abbiamo implementato